modules/openapi-generator/src/test/resources/3_0/allOf_extension_parent.yaml

```diff
@@ -0,0 +1,87 @@
+openapi: 3.0.1
+info:
+  version: 1.0.0
+  title: Example
+  license:
+    name: MIT
+servers:
+  - url: http://api.example.xyz/v1
+paths:
+  /person/display/{personId}:
+    get:
+      parameters:
+        - name: personId
+          in: path
+          required: true
+          description: The id of the person to retrieve
+          schema:
+            type: string
+      operationId: list
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Person"
+components:
+  schemas:
+    Person:
+      description: person using x-parent (abstrct) to indicate it's a parent class
+      type: object
+      x-parent: "abstract"
```

Comment:
I suggest a better name for this than abstract, maybe `interface`, since `abstract` implies a child object can have only 1 parent

Reply:
e.g. this schema https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/allOfMultiParent.yaml#L64-L83
the object has multiple `$ref` parents, what happens if one of them is `x-parent: true` and the other is `x-parent: false` ?

Reply:
another case is dart which supports mixins, so you can have
`class A extends B with C,D implements F,G ` will we have an `x-parent=abstract`  and `x-parent=mixin`  ?

Reply:
> I suggest a better name for this than abstract, maybe interface, since abstract implies a child object can have only 1 parent

`interface` is supported via `x-implements`

`abstract` is just an example here but coincidentally there's another PR to make parent `abstract` by default in the model.

The goal is eventually allow users to specify the parent type (e.g. `abstract`, `partial public`, etc)

The logic/implementation in this PR supports multiple parents (using $ref). This is same as before without this PR, which means we already support multiple parents if the languages/implementations supports it or default to just a single parent (first) if the language/implementation does not support multiple inheritance.